ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java

```diff
@@ -193,6 +193,13 @@ public Optional<EthPeer> bestPeerWithHeightEstimate() {
         p -> p.isFullyValidated() && p.chainState().hasEstimatedHeight());
   }
 
+  public Optional<EthPeer> bestPeerWithHeightEstimateForFastSync() {
+    return streamAvailablePeers()
+        .filter(p -> p.isFullyValidated() && p.chainState().hasEstimatedHeight())
+        .sorted(BEST_CHAIN)
```

Comment:
if we're taking max from downloadTime, we don't need to sort by best chain?

Reply:
Some peers may not have a download time recorded yet and the BEST_CHAIN is a fall back for when not enough peers have it

Reply:
what will the max() return in that case... tests would help!

Reply:
yep agreed, definitely needs to be min but I'll write tests with different scenarios.